addons/l10n_de/models/account_move.py

```diff
@@ -1,15 +1,12 @@
-from odoo import models, api
+from odoo import models, api, fields
 
 class AccountMove(models.Model):
     _inherit = 'account.move'
 
-    @api.depends('invoice_date')
-    def _compute_delivery_date(self):
-        # EXTENDS 'account'
-        super()._compute_delivery_date()
-        for move in self:
-            if move.invoice_date and move.country_code == 'DE' and not move.delivery_date:
-                move.delivery_date = move.invoice_date
+    @api.onchange('invoice_date')
+    def _onchange_delivery_date(self):
+        if self.invoice_date and self.country_code == 'DE' and not self.delivery_date:
+            self.delivery_date = self.invoice_date
```

Comment:
```suggestion
```
I agree with @adwid, I would also recommend to remove this extension.
The only concern is to set a delivery date when posting the invoice, just like the `invoice_date` actually.

We don't really care while the move is in draft.
If both stock and sale are installed, a default delivery date will be computed automatically (like in the provided use case for example):
https://github.com/odoo/odoo/blob/cf553a10d1bfccae775f1d7072db1521bfdff4f7/addons/sale_stock/models/account_move.py#L115-L124

We try to avoid using `@api.onchange()` as much as possible, and this is not really the kind of situations for which we should make an exception :nerd_face: